charts/redpanda/templates/statefulset.yaml

```diff
@@ -67,6 +67,16 @@ spec:
           volumeMounts:
             - name: datadir
               mountPath: /var/lib/redpanda/data
+{{- if and (or (not (empty .Values.license_key)) (not (empty .Values.license_secret_ref)) .Values.storage.tieredConfig.cloud_storage_enabled) (hasKey .Values.storage.tieredConfig "cloud_storage_cache_directory") (not (empty .Values.storage.tieredConfig.cloud_storage_cache_directory)) }}
```

Comment:
it will not fail if we will always have `cloud_storage_cache_directory` included in our `values.yaml` file. I'm not so used to it, so I will change it.

Reply:
I implemented the default template, so that this condition is now removed.